### PR TITLE
Maskable component

### PR DESCRIPTION
**Pull request by NehalBhandariSlalom**
This PR includes Maskable Component which renders an image. The image can be highlighted, selected and shows spinner until the image is loaded. This can be achieved by setting the appropriate properties.
Screen Shot 2021-11-09 at 3 10 21 PM

### CODE REVIEW

**Review by amirkibbar — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@toyota-research-institute/lakefront",
   "description": "React UI Components",
-  "version": "0.82.3",
+  "version": "0.82.5",
```
Comment (jason-ransom-slalom) [inline package.json:4]: Bump version to at least 0.83.0.
Btw: If you want the version to be bumped for you, you can include "feature" in at least one of your PRs (https://github.com/ToyotaResearchInstitute/lakefront/blob/main/CONTRIBUTING.md#versioning)

File: src/MaskableImage/assets/checkedOutline.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="#000000"><path d="M0 0h24v24H0V0z" fill="none"/><path d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41L9 16.17z"/></svg>
```
Comment (jason-ransom-slalom) [inline src/MaskableImage/assets/checkedOutline.svg:1]: nit: Add new line here.

File: src/MaskableImage/maskableImageStyles.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            fill: theme?.colors?.akoya,
+            borderRadius: '50%'
+        },
+        ...((!selectable || (!selected && (!hovered && !individuallySelected))) && { display: 'none' }),
+        ...(selectable && (selected || individuallySelected) && {
+            background: theme?.colors?.pastelGreen,
+            border: '1px solid', borderColor: theme?.colors?.akoya
+        })
+    })
+)
```
Comment (jason-ransom-slalom) [inline src/MaskableImage/maskableImageStyles.ts:74]: nit: Add new line here.

**Review by jason-ransom-slalom — CHANGES_REQUESTED** [review]
The component needs to be exported from the primary index file as well.

File: src/MaskableImage/MaskableImage.tsx [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    innerWidth?: number;
+}
+
+// This is the base64 for a 1x1 transparent pixel
+const transparentPixel = 'data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==';
+
+/**
+ * Maskable Image Component
+ *  
+ * This Component is used to render an image. The image can be selected and highlighted.The showSpinnerOnLoad when set
```
Comment (adorack-tri) [inline src/MaskableImage/MaskableImage.tsx:90]: Minor: there should be a space between '.' and 'The'.

File: src/MaskableImage/MaskableImage.tsx [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+        setIndividuallySelected(false);
+    }, [selected]);
+
+    // Swap the transparent pixel image for the real one once loaded
+    const handleImageLoaded = () => {
+        setImageLoaded(true);
+    };
+
+    // https://css-tricks.com/scaled-proportional-blocks-with-css-and-javascript/
+    // const aspectRatioConstraint = { paddingBottom: `${heightToWidthRatio * 100}%` };
```
Comment (adorack-tri) [inline src/MaskableImage/MaskableImage.tsx:173]: Please remove dead code.

**Review by adorack-tri — CHANGES_REQUESTED** [review]
For the image (nature.jpg), I think it would make sense to reduce the quality in order to reduce the file size.  I'll send an example via Slack.

**Review by jason-ransom-slalom — APPROVED** [review]

**Comment by NehalBhandariSlalom** [comment]
> For the image (nature.jpg), I think it would make sense to reduce the quality in order to reduce the file size. I'll send an example via Slack.

I have uploaded a lower quality image. Please check if this looks good.

**Review by adorack-tri — APPROVED** [review]

**Pull request merged**